### DISCUSSION TITLE
Accidentally unlinked all commits, need help linking them back.

### DISCUSSION THREAD

**Discussion by Beelijah** [community/community#144098] [Other Feature Feedback, Questions, & Ideas]
# what's the repository you're talking about?
[this one right here](https://github.com/Beelijah/fnfcleanweb.github.io)

# what's your project about?
woah woah woah, i didn't make this!
its just a mod i made for [Friday Night Funkin'](https://github.com/FunkinCrew/Funkin)

# what did you do
fricked around with my .git folder. i saw it taking up a ton of space and was like "wow, im deleting this junk!".
little did i know, this would cost me 5 hours of my life.

# so what happened
all 71 commits ive made are unlinked. they don't show up under commits, and when viewing them (found them in the deployments page), for example [this one](https://github.com/Beelijah/fnfcleanweb.github.io/commit/94ad08e000f2c4b48531e6b337324ce293c76f6d) will give a warning at the top, stating "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.".

# how to help you fix?
we need to, in a way, link the 71 commits back to a branch, which is main.
and how do we do that? well, i don't know. ive sat for about an hour just chatting to AI about how to do this, and nothing has worked.

HELP NEEDED!

**Comment by samus-aran** [comment]
Oft. 😬 Did you completely got rid of your .git folder? Can you restore it from your machine (like checking the wastebin?). That would be the easiest way.

Alternatively you will need to source [another local copy that has your commits in it](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/commit-exists-on-github-but-not-in-my-local-clone#the-branch-that-contained-the-commit-was-deleted) and try to recover it from that.

Question: Is it a matter that you want these commits to appear associated to your account or that it impacts you accessing them if you re-clone the repo again?

**Comment by Beelijah** [comment]
Completely gone. Shift + Delete. Plus, its been a few days so I'm pretty sure stuff like Recuva wont work.
I did have a local copy, but when checking the data it was very corrupted.

They ARE all there, just not linked to any branch.

Answer: I just want the commits to show up on the branch.

**Comment by HughDavenport111** [comment]
Looks like everyone’s gotten used to turning to AI for help when they have questions,

**Reply by Beelijah** [reply]
ye because _most of the time_ its actually pretty useful!

**Comment by HughDavenport111** [comment]
SO AM I..
